paths_cli/parameters.py

```diff
@@ -18,7 +18,29 @@
     store='schemes',
 )
 
-INIT_CONDS = OPSStorageLoadMultiple(
+class InitCondsLoader(OPSStorageLoadMultiple):
+    def _extract_trajectories(self, obj):
+        import openpathsampling as paths
+        if isinstance(obj, paths.SampleSet):
+            yield from (s.trajectory for s in obj)
+        elif isinstance(obj, paths.Sample):
+            yield obj.trajectory
+        elif isinstance(obj, paths.Trajectory):
+            yield obj
+        elif isinstance(obj, list):
+            for o in obj:
+                yield from self._extract_trajectories(o)
+        else:
+            raise RuntimeError("Unknown initial conditions type: "
+                               f"{obj} (type: {type(obj)}")
+
+    def get(self, storage, names):
+        results = super().get(storage, names)
+        final_results = list(self._extract_trajectories(results))
```

Comment:
For my sanity, if the return from storage is `paths.Trajectory`, this returns [`traj`] and not `[snap0, snap1, snap2]` 
There's no implicit list casting supported for `paths.Trajectory`, right?

Reply:
Yes, this is correct. In all cases, the goal here is to make this return a list of trajectories.

There is some implicit casting in OPS, but not from `list[Snapshot]` to `Trajectory`. An example of why is when we calculate the max-lambda in a TIS trajectory. We store a mapping of trajectory UUID to result -- a list of snapshots doesn't have a UUID.

The other way works, however: a CV that receives a trajectory will act the same as if it received a list of snapshots. Similarly, things are pretty clever (maybe too much so?) about allowing a `Sample` to stand in for a `Trajectory`.

I think key to your question here is this:

https://github.com/openpathsampling/openpathsampling-cli/blob/ff15749584c829f86d4d482b56943ff124e79d8b/paths_cli/parameters.py#L28-L29

We're in a generator, so if `results` is a single trajectory, then we yield that trajectory and soon raise StopIteration. So `list(self._extract_trajectories)` creates a list containing a single trajectory.

There is also some (too?) clever casting stuff in `initial_conditions_from_trajectories` to try to figure out user intent. It allows a trajectory or pretty much any container of trajectories. However, I kept trying to feed it a container of containers of trajectories, hence this PR.